sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -689,27 +722,45 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
         private async Task RunReceiveTaskAsync(
             CancellationToken cancellationToken)
         {
-            List<Task> tasks = new List<Task>(_maxConcurrentCalls + _receiverManagers.Count);
             try
             {
                 while (!cancellationToken.IsCancellationRequested && !Connection.IsClosed)
                 {
+                    ReconcileConcurrency();
+                    _handlerCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
                     foreach (ReceiverManager receiverManager in _receiverManagers)
                     {
                         // Do a quick synchronous check before we resort to async/await with the state-machine overhead.
                         if (!_messageHandlerSemaphore.Wait(0, CancellationToken.None))
```

Comment:
Is it OK that we won't pick up the update until we feel all active slots in the semaphore?

Reply:
What do you mean - this has a timeout of 0.

Reply:
Right, so we would use up all semaphore slots before we get to WaitAsync that can be canceled by Update***

Reply:
Ah fair point.